usbd-storage/src/subclass/scsi.rs

```diff
@@ -40,14 +41,34 @@ const WRITE_10: u8 = 0x2A;
 /* MMC */
 const READ_FORMAT_CAPACITIES: u8 = 0x23;
 
+/// A u8, when printed shown as hex
+#[repr(transparent)]
+#[derive(Copy, Clone)]
+pub struct U8Hex(pub u8);
```

Comment:
I like the idea of printing in hex, however I would expand it to all `ScsiCommand`s and `UfiCommand`s and therefore for `u16`, `u64` and `u32`.
The newtype is an obvious choice but I wonder if we could rely on https://defmt.ferrous-systems.com/hints#propagation instead. If I understand it correctly, then by adding something like `:#X` to `src/fmt.rs` macros, we get Hex formatting everywhere for free?

I'm not sure that `derive(defmt::Format)` would add `=u8`-like typing info, so maybe it would require an explicit implementation of `defmt::Format` for `ScsiCommand` and `UfiCommand`? (may put  them in `fmt.rs`)

As a last resort, I would still try to implement newtypes for other primitives and see how much of a code change it requires. (I don't mind changing a public API. that's what semver is for :) )

Reply:
I don't think propagation does help, as entries with display hint (= show hex) can't be overwritten. Also only a struct (or so) can implement Format, but since the command is a u8 it can't be formatted without newtype.

Yes, I should add =u8, and probably use `{=u8:#04x}` instead of `0x{:02x}`.

Reply:
I tried 
```rust
#[derive(defmt::Format)]
enum FakeCommand {
    Unknown {
        u16: u16,
        u32: u32,
        u64: u64,
        u128: u128,
    }
}

#[cortex_m_rt::entry]
fn main() -> ! {
    let foo = FakeCommand::Unknown{
        u16: 0xCAFE,
        u32: 0xCAFE,
        u64: 0xCAFE,
        u128: 0xCAFE,
    };
    defmt::info!("{:#X}", foo);
}
```
and it printed
`Unknown { u16: 0xCAFE, u32: 0xCAFE, u64: 0xCAFE, u128: 0xCAFE }`.

So if `:#X` hints are added to `fmt.rs` macros, then crate-internal logging will output everything in hex.

Note, that this does not currently work with `u64` and `u128`. Hopefully https://github.com/knurling-rs/defmt/pull/869 will be merged